### PR TITLE
test: Added Atoms & components Test cases

### PR DESCRIPTION
**Pull request by nayanmarlin**
**### Atoms:**

- [x] Tooltips

- [x] Toast

- [x] Timer

- [x] Tabs

- [x] Tooltip icon

- [x] Tabs Panel

- [x] Tab Lists

- [x] Tab

**### Components:**

- [x] Wrapper

- [x] Texts

- [x] Table Cells

- [x] Status

- [x] Select

- [x] Modal

- [x] Images

- [x] HeaderLinksGroup

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
 		"eslint": "^8.48.0",
 		"eslint-config-prettier": "^9.0.0",
 		"eslint-plugin-svelte": "^2.33.0",
-		"jsdom": "^23.0.1",
+		"jsdom": "^23.2.0",
```
Comment (vishwajeetraj11) [inline package.json:32]: is this necessary?

File: src/lib/atoms/timer/TestTimer.svelte [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
```
Comment (vishwajeetraj11) [inline src/lib/atoms/timer/TestTimer.svelte:1]: why is there a need of testTimer?

File: package.json (hunk 1 above)
Comment (nayanmarlin) [reply package.json]: It was automatically added while running test case

Comment (nayanmarlin) [reply package.json]: Removed.

File: src/lib/atoms/timer/TestTimer.svelte (hunk 2 above)
Comment (nayanmarlin) [reply src/lib/atoms/timer/TestTimer.svelte:1]: To get <slot /> element @vishwajeetraj11

File: src/lib/atoms/tooltips/TooltipIcon.dom.spec.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import { staticImages } from '$lib/components/images/staticImages';
+
+describe('Tooltip Icon', () => {
+	afterEach(() => {
+		cleanup();
+		vi.resetAllMocks();
+	});
+
+	test('tooltipIcon renders with default attribute', () => {
+		const { getByTestId } = render(TooltipIcon, { props: { tooltipText: '' } });
```
Comment (souvikmishra) [inline src/lib/atoms/tooltips/TooltipIcon.dom.spec.ts:12]: maybe use a specific string to assert data tip content

File: src/lib/atoms/tabs/Tabs.dom.spec.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	});
+
+	test('tab Render', () => {
+		const { getByTestId } = render(Tab, { props: { id: 'tab_test' } });
+		const tabElement = getByTestId('tab');
+		expect(tabElement.childNodes).toHaveLength(1);
+		expect((tabElement.firstChild as Element)?.id === 'tab_test-tabhead').toBe(true);
+	});
+
+	test('tab Render check using getByTestId & getByRole ', () => {
```
Comment (souvikmishra) [inline src/lib/atoms/tabs/Tabs.dom.spec.ts:18]: whitespace inconsistency in mulitple places, maybe use prettier to format

File: src/lib/atoms/timer/TestTimer.svelte (hunk 2 above)
Comment (souvikmishra) [inline src/lib/atoms/timer/TestTimer.svelte:1]: we can use `svelte-htm` for this probably, @vishwajeetraj11 has used it in his previous PRs to test slots

File: src/lib/atoms/timer/Timer.dom.spec.ts [hunk 5]
```diff
@@ -1,28 +1,103 @@
...
+import Timer from './Timer.svelte';
+import TestTimer from './TestTimer.svelte';
+
+const timerTestId = 'timer-test';
+const delay = (duration: number) => new Promise((resolve) => setTimeout(resolve, duration));
+
+describe('TimerComponent Component', () => {
+
+	beforeAll(() => {
+		const container = document.createElement('div');
```
Comment (souvikmishra) [inline src/lib/atoms/timer/Timer.dom.spec.ts:12]: why are we creating a div here?

File: src/lib/atoms/timer/Timer.dom.spec.ts [hunk 6]
```diff
@@ -1,28 +1,103 @@
...
+	});
+
+	afterEach(() => {
+		cleanup();
+		vi.resetAllMocks();
+	});
+
+	test('renders without crashing', () => {
+		const { container } = render(Timer);
+		expect(container.firstChild).toBeTruthy();
```
Comment (souvikmishra) [inline src/lib/atoms/timer/Timer.dom.spec.ts:23]: we can probably use a data test id or a selector which gives us confidence about the rendered element

File: src/lib/atoms/timer/Timer.svelte [hunk 7]
```diff
@@ -1,7 +1,6 @@
 <script lang="ts">
 	import { doNothing } from '$lib/utils/helpers/commonHelper';
-	import { onDestroy } from 'svelte';
-
+
```
Comment (souvikmishra) [inline src/lib/atoms/timer/Timer.svelte:3]: formatting issue

File: src/lib/atoms/tooltips/Tooltip.dom.spec.ts [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+	afterEach(() => {
+		cleanup();
+		vi.resetAllMocks();
+	});
+
+	test('tooltip renders with default attribute', () => {
+		const { getByTestId } = render(Tooltip, { props: { tooltipText: '' } });
+		const tooltipElement = getByTestId('tooltip');
+		const tooltipClasslist = tooltipElement.classList;
+		expect(tooltipElement.getAttribute('data-tip')).toEqual('');
```
Comment (souvikmishra) [inline src/lib/atoms/tooltips/Tooltip.dom.spec.ts:14]: maybe use a specific string to assert content of data-tip

File: src/lib/atoms/tabs/Tabs.dom.spec.ts (hunk 4 above)
Comment (nayanmarlin) [reply src/lib/atoms/tabs/Tabs.dom.spec.ts:18]: I am already using prettier @souvikmishra , And I crossed check with this file, there is no changes to be format in this file. by the way is tab space issue? how many tab space size you set in your VS Code?

File: src/lib/atoms/timer/Timer.dom.spec.ts (hunk 5 above)
Comment (nayanmarlin) [reply src/lib/atoms/timer/Timer.dom.spec.ts]: I was testing earlier, forget to remove. Removed in latest commit.

**Comment by nayanmarlin** [comment]
Due to sort of limitation, **`Header.svelte`** This file's UTs are not covered.
Check out more details here: https://stackoverflow.com/questions/77070392/how-to-mock-svelte-store-with-vitest

CC: @souvikmishra

**Review by souvikmishra — APPROVED** [review]

**Pull request merged**